spec/21-http_header_format_rationale.md

```diff
@@ -16,6 +16,103 @@ Making `trace-flags` optional doesn't save a lot, but makes specification more c
+
+Thus for improved interoperability, this specification defines `trace-id` as a
+128 bit array of bytes.
+
+## Trace ID and interoperability with 64bit systems
+
+There are systems today using 64-bit `trace-id`s. These systems are not always
+easy to switch to longer `trace-id` to confirm to this specification
+requirement. The cost of this switch can be prohibitively expensive from both -
+backend capacity and indexes as well as in-process propagation limits.
+
+When addressing interoperability with these systems requirements the following
+is taken into consideration:
+
+1. The main objective of the specification is to promote interoperability of
+   various vendors and platforms.
+2. Specification needs to suggest a best practices that will improve
+   interoperability.
+3. There must be a way forward to implement Trace Context protocol by systems
+   that doesn't support long trace identifiers.
+
+### How 64bit systems may switch to Trace Context
+
+#### Absolute minimum
+
+The absolute minimum requirement from specification perspective is to receive
+and send a valid `traceparent` header. Systems operating with shorter `trace-id`
+may use only a subset of `trace-id` bytes to read and set `trace-id`. This
+behavior will break interoperability with vendors and platforms operating with
+the longer identifiers. If only a subset of `trace-id` bytes were read on
+incoming requests and sent with the outgoing call, for the systems operating
+with longer identifiers incoming and outgoing `trace-id` will not match. These
+systems will identify this situation as a restarted trace.
+
+Note, that vendors and platforms may implement a special logic to interoperate
+with the systems like this.
+
+Specification uses "SHOULD language" when asking to fill up extra bytes with
+random numbers instead of zeros. Typically, tracing systems that will not
+propagate extra bytes of incoming `trace-id` will not follow this ask and will
+fill up extra bytes with zeros.
+
+#### Linking to the longer thread-id
+
+As a minor step to improve interoperability between tracing systems, system that
+operates with shorter identifiers may record a longer incoming `trace-id` as a
+property of a telemetry item representing the incoming request.
+
+#### Propagating extra bytes of trace-id
+
+Preserving the `trace-id` unchanged is a major improvement in interoperability
+of tracing systems using different number of `trace-id` bytes. It is typical
+that tracing systems may propagate extra information from incoming request to
+outgoing calls. While recording of these extra bytes to the tracing systems
+backend is not possible.
+
+If tracing system can propagate these extra bytes, than it MUST do it.
+
+When tracing system has this capability, specification suggests that extra bytes
+are fill out with random numbers instead of zeros. This requirement helps to
+validate that tracing systems implements propagation from incoming request to
```

Comment:
What use is this validation? Isn't it better to have de-facto interoperability by default? Why is this validation more beneficial and outweigh default interoperability with existing brown field systems?

Reply:
I think the key here is that spec is addressing interoperabilty of different tracing systems. So if 64 bit system receives 128bit trace-id, it must do the best effort to propagate the entire 128 bits. The best way to ensure that this is implemented is ask to fill up spaces with random numbers.

Reply:
No, actual users are telling you that the best way to ensure that is to backfill with zeros.